pkg/streamingpromql/query.go

```diff
@@ -502,7 +522,15 @@ func (q *Query) Exec(ctx context.Context) *promql.Result {
 		} else {
 			q.result = &promql.Result{Value: q.populateMatrixFromScalarOperator(d)}
 		}
-
+	case parser.ValueTypeString:
+		if q.IsInstant() {
```

Comment:
We don't bother checking this for range vectors above - I don't think this is necessary here, as it should be caught in `newQuery` above.

Reply:
Let's leave it for now.